pkg/pillar/cmd/diag/diag.go

```diff
@@ -542,20 +722,98 @@ func handleOnboardStatusImpl(ctxArg interface{}, key string,
 	}
 	ctx.devUUID = status.DeviceUUID
 	log.Functionf("handleOnboardStatusImpl changed to %v", ctx.devUUID)
-	printOutput(ctx)
+	triggerPrintOutput(ctx, "Onboard")
+}
+
+// Conditional send to ensure that one gets printed but multiple triggers
+// can collapse into one print
+func triggerPrintOutput(ctx *diagContext, caller string) {
+	select {
+	case ctx.triggerPrintChan <- caller:
+		// Do nothing more
+	default:
+	}
+}
+
+// printTask waits for 5 seconds after each print to limit the rate
+func printTask(ctx *diagContext, triggerPrintChan <-chan string) {
+	for {
+		select {
+		case caller := <-triggerPrintChan:
+			printOutput(ctx, caller)
+			time.Sleep(5 * time.Second)
+		}
+	}
 }
 
 // Print output for all interfaces
 // XXX can we limit to interfaces which changed?
-func printOutput(ctx *diagContext) {
+func printOutput(ctx *diagContext, caller string) {
+
+	lines := 0 // Count numer of Fprintf calls
+	fmt.Fprintf(outfile, "\nINFO: updated diag information at %v due to %s\n",
+		time.Now().Format(time.RFC3339Nano), caller)
+	lines++
+	// Determine what to print for the device line
+	printErr := false
+	level := "INFO"
+	if ctx.zedagentStatus.VaultStatus == info.DataSecAtRestStatus_DATASEC_AT_REST_DISABLED {
+		level = "WARNING" // Can be overridden below
+	}
+	if ctx.zedagentStatus.DeviceState == types.DEVICE_STATE_MAINTENANCE_MODE {
+		level = "ERROR"
+	} else if ctx.zedagentStatus.AttestError != "" {
+		level = "ERROR"
+		printErr = true
+	} else if ctx.zedagentStatus.VaultErr != "" &&
+		ctx.zedagentStatus.VaultStatus != info.DataSecAtRestStatus_DATASEC_AT_REST_DISABLED {
+		level = "ERROR"
+		printErr = true
+	}
+
+	if printErr {
+		fmt.Fprintf(outfile, "%s: device %s attestation %s error %s vault %s PCR %s error %s\n",
+			level,
+			ctx.zedagentStatus.DeviceState.String(),
+			ctx.zedagentStatus.AttestState.String(),
+			ctx.zedagentStatus.AttestError,
+			ctx.zedagentStatus.VaultStatus.String(),
+			ctx.zedagentStatus.PCRStatus.String(),
+			ctx.zedagentStatus.VaultErr)
+		lines++
```

Comment:
I wonder if we could avoid having to put `lines++` for every Fprintf. Perhaps the function could collect output to a buffer first, then measure output (decide what to do if it exceeds the window size) and then print it to outfile.
Something like:
```
var b bytes.Buffer
fmt.Fprintf(&b, "world!")
...
lines := bytes.Count(buffer.Bytes(), []byte{'\n'}) + 1
b.WriteTo(outfile)
```

Reply:
@milan-zededa addd the above check (still needs to test that stty size is correct on a praphical console). Thanks for the suggestion.